File: fixtures/source/rust/audit_store.rs
pub struct AuditEvent {
    pub id: String,
}

pub struct AuditRepo;

impl AuditRepo {
    pub fn insert(&self, _event: AuditEvent) {}
}

pub struct AuditStore {
    repo: AuditRepo,
}

impl AuditStore {
    pub fn append_event(&self, event: AuditEvent) {
        self.repo.insert(event);
    }
}

File: fixtures/source/audit_store.ts
export interface AuditEvent {
  id: string;
}

export class AuditStore {
  repo: { insert: (event: AuditEvent) => unknown };

  constructor(repo: { insert: (event: AuditEvent) => unknown }) {
    this.repo = repo;
  }

  appendEvent(event: AuditEvent) {
    return this.repo.insert(event);
  }
}
